backend/appwrite-functions/announcement/ancm-function/main.py

```diff
@@ -0,0 +1,212 @@
+    # Success message
+    print(
+        json.dumps(
+            {
+                "action": "removeAnnouncements",
+                "status": "success",
+                "sum": len(removed),
+                "removed": removed,
+            }
+        )
+    )
+
+
+def handleGet(client_payload: dict, database: Database, collection_id: str):
+    # Extract data from payload
+    nbr_ancm = client_payload["numberOfAnnouncements"]
+    timestamp = client_payload["timestamp"]
+    after = client_payload["after"]
+    # Prepare payload to return to client
+    return_payload = {
+        "action": "getAnnouncements",
+        "status": "success",
+        "sum": -1,
+        "announcements": [],
+    }
+    # Query data from collection
+    listDocuments = database.list_documents(
+        collection_id=collection_id,  # collection["$id"],
+        order_field="created_at",
+        order_type="DESC",
+        filters=[f"created_at>={timestamp}"] if after else [f"created_at<={timestamp}"],
+        limit=nbr_ancm,
+    )
+    # Add data to payload
+    return_payload["sum"] = len(listDocuments["documents"])
+    for document in listDocuments["documents"]:
+        return_payload["announcements"].append(
+            {
+                "created_at": document["created_at"],
+                "updated_at": document["updated_at"],
+                "title": document["title"],
+                "content": document["content"],
+            }
+        )
+    # There is no way to return JSON data back directly to client, so we transfer by using stdout.
+    print(json.dumps(return_payload))
+    exit()
+
+
+def main():
+    client = init_client()
+    database = Database(client)
+    PAYLOAD = os.environ.get("APPWRITE_FUNCTION_DATA")
+    if not PAYLOAD:
+        exit()
+    try:
+        client_payload = json.loads(PAYLOAD)
+        if "action" not in client_payload:
+            print_failure_message("'action' key not found in data object!")
+            exit()
+        collection_id = get_collection_id(database=database)
+
+        if client_payload["action"] == "getAnnouncements":
+            handleGet(
+                client_payload=client_payload,
+                database=database,
+                collection_id=collection_id,
+            )
+        elif client_payload["action"] == "addAnnouncements":
+            handleAdd(
+                client_payload=client_payload,
+                database=database,
+                collection_id=collection_id,
+            )
+        elif client_payload["action"] == "updateAnnouncements":
+            handleUpdate(
+                client_payload=client_payload,
+                database=database,
+                collection_id=collection_id,
+            )
+        elif client_payload["action"] == "removeAnnouncements":
+            handleRemove(
+                client_payload=client_payload,
+                database=database,
+                collection_id=collection_id,
+            )
```

Comment:
I was thinking about this:
```suggestion
        handleAnnouncement = (
            getAnnouncement if client_payload["action"] == "getAnnouncements" else
            addAnnouncement if client_payload["action"] == "addAnnouncements" else
            updateAnnouncement if client_payload["action"] == "updateAnnouncements" else
            removeAnnouncement if client_payload["action"] == "removeAnnouncements" else
            lambda x, y, z: pass
        )

            handleAnnouncement(
                client_payload=client_payload,
                database=database,
                collection_id=collection_id,
            )
```

Reply:
I already had 21 commits. Might consider next time.

Reply:
I'm sorry, yes, it likely took you a lot of time. According to review guidelines cloned code like this should be pointed out. I wouldn't be so pesky about it if this wouldn't be an open source project.

Reply:
lets vote: @Floinko  @dominic-heil

Reply:
my best idea: close this old issue, and create "improvement issue". I was stuck with this one for way too long.

Reply:
I do not see a reason for too many changes here as this function will not be used in production and will only be a reference for the future.

Reply:
Ah, okay. If that's the case I won't care so much anymore.